quipucords/scanner/openshift/api.py

```diff
@@ -362,6 +362,13 @@ def _list_routes(self, **kwargs):
         return self._route_api.get(**kwargs).items
 
     def _init_ocp_nodes(self, node) -> OCPNode:
+        unschedulable = node.spec["unschedulable"]
+        if unschedulable is None:
+            # following upstream docs[1], if unschedulable is None, it is considered
+            # false.
+            # [1]: https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/node-v1/#NodeSpec
+            unschedulable = False
```

Comment:
How much do we trust the `spec` dict to be well-structured?
```suggestion
        # following upstream docs[1], if unschedulable is None, it is considered false.
        # [1]: https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/node-v1/#NodeSpec
        unschedulable = node.spec.get("unschedulable", False) or False
```

Reply:
spec isn't a dictionary. This is a magical `ResourceField`. which is a weird magical object that always returns keys, even when they don't exist.

If I ask node["foo"]["bar"] it will just return `None` 🤢 . This implies that if "unschedulable" isn't set (which is the default), `node.spec.get("unschedulable", False) ` would be equal to `None` 🤷🏽 .

TBH, I hate it. As I hate the lib we use for interacting with OCP (just adopted it because interacting directly with the API would take way more time to implement). While this kind of "smart pants" object can be neat to use, it is a huge pain to test (see the disgusting workarounds to create the object to test this).

Part of the reason I adopted vcr to test the integration with ocp was this type of thing.

Reply:
I changed it anyways. See the fixup commit.
However, if we don't trust Openshift/kubernetes response, we can't properly infer `unschedulable`.

But if we go that far and don't trust the response, maybe we should not trust `spec` will be present as well.

At that point I don't see a distinction between this line of flexibility and network scan `ignore_errors: true` on all ansible tasks.